packages/slate-react/src/components/leaf.js

```diff
@@ -60,9 +60,9 @@ class Leaf extends React.Component {
     // If any of the regular properties have changed, re-render.
     if (
       props.index != this.props.index ||
-      props.marks != this.props.marks ||
+      !props.marks.equals(this.props.marks) ||
       props.text != this.props.text ||
-      props.parent != this.props.parent
```

Comment:
What are these changes for?

Reply:
Because marks can be the same value with different reference after characters-to-leaves.  Therefore I think it is perhaps better with immutable equals